lms/services/http.py

```diff
@@ -0,0 +1,161 @@
+import requests
+from requests import RequestException
+
+from lms.services.exceptions import HTTPError, HTTPValidationError
+from lms.validation import ValidationError
+
+
+class HTTPService:
+    """Send HTTP requests with `requests` and receive the responses."""
+
+    def __init__(self, _session=None):
+        # A requests session is used so that cookies are persisted across
+        # requests and urllib3 connection pooling is used (which means that
+        # underlying TCP connections are re-used when making multiple requests
+        # to the same host, e.g. pagination).
+        #
+        # See https://docs.python-requests.org/en/latest/user/advanced/#session-objects
+        self._session = _session or requests.Session()
+
+    def request(
+        self,
+        method,
+        url,
+        params=None,
+        data=None,
+        json=None,
+        headers=None,
+        auth=None,
+        timeout=9,
```

Comment:
Maybe a tuple is a better default here (connect_timeout, read_timeout)

https://2.python-requests.org/en/master/user/advanced/#timeouts

Reply:
Yeah, it might be better for readability to do `timeout=(10,10)` as it just clarifies, from the default value, that there are actually two separate timeouts not one.

Would we want the two timeouts to have different values?

I also wonder whether we should change the default from `9` to `10` because the requests docs say:

>  It’s a good practice to set connect timeouts to **slightly larger than a multiple of 3**, which is the default TCP packet retransmission window.
> 
> --- https://docs.python-requests.org/en/master/user/advanced/#timeouts

Reply:
Done: https://github.com/hypothesis/lms/pull/2711/commits/d54c0796f3b5b15119b7b59ea00dadff0ce54f1e